internal/controller/ledger/numscript_parser.go

```diff
@@ -33,7 +33,7 @@ func NewDefaultNumscriptParser() *DefaultNumscriptParser {
 
 var _ NumscriptParser = (*DefaultNumscriptParser)(nil)
 
-type InterpreterNumscriptParser struct{}
+type InterpreterNumscriptParser struct{ featureFlags map[string]struct{} }
```

Comment:
Why not using a string slice?

Reply:
the interpreter eventually takes a `map[string]struct{}`
I though having a set would be a more consistent data structure for an unordered collection of unique items
..except we don't have a set in go so we use that map thing

I could also carry a slice around and finally convert it to a map but for the same reason, I though it's better not to pass around unparsed, possibly inconsistent data